packages/runtime-core/src/hydrationStrategies.ts

```diff
@@ -41,19 +41,20 @@ export const hydrateOnVisible: HydrationStrategyFactory<
 > = opts => (hydrate, forEach) => {
   const ob = new IntersectionObserver(entries => {
     for (const e of entries) {
-      if (!e.isIntersecting) continue
-      ob.disconnect()
-      hydrate()
-      break
+      if (e.isIntersecting) {
+        hydrate()
+        ob.disconnect()
+        return
+      }
     }
   }, opts)
   forEach(el => {
     if (elementIsVisibleInViewport(el)) {
       hydrate()
       ob.disconnect()
-      return false
```

Comment:
The `return false` here has a special purpose
see https://github.com/vuejs/core/pull/11639#discussion_r1756426473